### PR TITLE
[WIP] ci: run tests in safari

### PR DESCRIPTION
**Pull request by straker**
Discovered we didn't run any of our code through Safari in CircleCi. Huge thanks to @michael-siek for helping with this pr.

## Reviewer checks

**Required fields, to be filled out by PR reviewer(s)**

- [ ] Follows the commit message policy, appropriate for next version
- [ ] Code is reviewed for security

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -30,6 +34,12 @@ restore_dependency_cache_win: &restore_dependency_cache_win
       - v8-cache-win-{{ checksum "package.json" }}
       - v8-cache-win-
 
+restore_dependency_cache_macos: &restore_dependency_cache_macos
+  restore_cache:
+    keys:
+      - v8-cache-win-{{ checksum "package.json" }}
```
Comment (stephenmathieson) [inline .circleci/config.yml:40]: Should this be a MacOS specific cache?

File: .circleci/config.yml [hunk 2]
```diff
@@ -56,6 +66,18 @@ jobs:
...
+  dependencies_macos:
+    <<: *defaults
+    <<: *macos_box
+    steps:
+      - checkout
+      - <<: *set_npm_auth
+      - <<: *restore_dependency_cache_macos
+      - run: npm ci
+      - save_cache:
+          key: v8-cache-unix-{{ checksum "package.json" }}
```
Comment (stephenmathieson) [inline .circleci/config.yml:78]: Don't think we should clobber the Unix cache here.

File: .circleci/config.yml [hunk 3]
```diff
@@ -30,6 +34,12 @@ restore_dependency_cache_win: &restore_dependency_cache_win
       - v8-cache-win-{{ checksum "package.json" }}
       - v8-cache-win-
 
+restore_dependency_cache_macos: &restore_dependency_cache_macos
+  restore_cache:
+    keys:
+      - v8-cache-win-{{ checksum "package.json" }}
+      - v8-cache-win-
```
Comment (stephenmathieson) [inline .circleci/config.yml:41]:
```suggestion
      - v8-cache-macos-{{ checksum "package.json" }}
      - v8-cache-macos-
```

File: .circleci/config.yml (hunk 2 above)
Comment (stephenmathieson) [inline .circleci/config.yml:78]:
```suggestion
          key: v8-cache-macos-{{ checksum "package.json" }}
```

File: .circleci/config.yml [hunk 4]
```diff
@@ -83,23 +104,45 @@ jobs:
...
+        - run: npm run build
+        # get fixtures ready for running tests
+        - run: npx grunt testconfig
+        - run: npx grunt fixture
+        # enable connection to safaridriver
+        - run: |
+            defaults write com.apple.Safari IncludeDevelopMenu YES
+            defaults write com.apple.Safari AllowRemoteAutomation 1
+            sudo safaridriver --enable
+            safaridriver -p 0 &
```
Comment (stephenmathieson) [inline .circleci/config.yml:142]: This should be a ["background" command](https://circleci.com/docs/2.0/configuration-reference/#background-commands).

```suggestion
        - run: |
            defaults write com.apple.Safari IncludeDevelopMenu YES
            defaults write com.apple.Safari AllowRemoteAutomation 1
            sudo safaridriver --enable
        - run:
              command: safaridriver -p 0
              background: true
```

File: .circleci/config.yml [hunk 5]
```diff
@@ -83,23 +104,45 @@ jobs:
...
+        # run safari webdriver tests
+        - run: npx grunt connect test-webdriver:safari --force
```
Comment (stephenmathieson) [inline .circleci/config.yml:144]: What is `--force` doing here?

File: build/tasks/test-webdriver.js [hunk 6]
```diff
@@ -181,15 +177,7 @@ module.exports = function(grunt) {
...
-			driver
-				.manage()
-				.timeouts()
-				.setScriptTimeout(!isMobile ? 60000 * 5 : 60000 * 10);
-			// allow to wait for page load implicitly
-			driver
-				.manage()
-				.timeouts()
-				.implicitlyWait(50000);
+			driver;
```
Comment (stephenmathieson) [inline build/tasks/test-webdriver.js:180]: What is this for?

**Comment by WilcoFiers** [comment]
This doesn't seem to be any sort of priority. Closing it until we decide to pick this up again.

**Pull request closed without merging**